packages/lib/modules/web3/WagmiConfig.ts

```diff
@@ -24,44 +24,48 @@ import { isBrowser } from '@repo/lib/shared/utils/app.helper'
 const walletConnectProjectId = process.env.NEXT_PUBLIC_WALLET_CONNECT_ID
 if (!walletConnectProjectId) throw new Error('Missing NEXT_PUBLIC_WALLET_CONNECT_ID env')
 
-const connectors = connectorsForWallets(
-  [
+let connectors: ReturnType<typeof connectorsForWallets> = []
+
+if (isBrowser()) {
+  connectors = connectorsForWallets(
+    [
+      {
+        groupName: 'Recommended',
+        wallets: [
+          // metaMaskWallet must appear above injectedWallet to avoid random disconnection issues
+          metaMaskWallet,
+          safeWallet,
+          rabbyWallet,
+          coinbaseWallet,
+          rainbowWallet,
+          injectedWallet,
+        ],
+      },
+    ],
     {
-      groupName: 'Recommended',
-      wallets: [
-        // metaMaskWallet must appear above injectedWallet to avoid random disconnection issues
-        metaMaskWallet,
-        safeWallet,
-        rabbyWallet,
-        coinbaseWallet,
-        rainbowWallet,
-        injectedWallet,
-      ],
-    },
-  ],
-  {
-    appName: PROJECT_CONFIG.projectName,
-    projectId: walletConnectProjectId,
-  }
-)
+      appName: PROJECT_CONFIG.projectName,
+      projectId: walletConnectProjectId,
+    }
+  )
 
-/*
+  /*
   Only adding a new WC Connector if the user is not already connected to WC
   This fixes this rainbowkit issue:
   https://github.com/rainbow-me/rainbowkit/issues/2232
 */
-if (isBrowser() && !isConnectedToWC()) {
-  const lastConnector = connectors[connectors.length - 1]
-  if (lastConnector({} as any).id !== 'walletConnect') {
-    connectors.push(
-      createWalletConnectConnector({ index: connectors.length, walletConnectProjectId })
-    )
+  if (!isConnectedToWC()) {
+    const lastConnector = connectors[connectors.length - 1]
```

Comment:
Is this file executed on the server and the browser? I see that it has the 'use client' directive, but I am not sure if that makes any sense outside components and hooks. If it only executes on the browser, is the conditional above needed? If it executes in both envs, wouldn't `lastConnector` be undefined in the server case and we will have some kind of error in the next line?

Reply:
no just for the wagmi config

```
export const wagmiConfig = createConfig({
  chains,
  transports,
  connectors,
  ssr: true, // <- set to 'false'
})
```

Reply:
apparently but it's only needed on the client